sophie-kaleba/temporary_assignment5/Alphabet.java

```diff
@@ -0,0 +1,179 @@
+		Map<Character, VertexState> marked = new LinkedHashMap<Character, VertexState>();
+		
+		for (Character c : unsortedAlphabet.getContent().keySet()) {
+			marked.put(c, VertexState.WHITE);
+		}
+		
+		for (Character c : marked.keySet()) {
+			if (marked.get(c) == VertexState.WHITE) {
+				this.depthFirstSearch(unsortedAlphabet, marked, stack, c);
+			}
+		}
+		
+		while (!stack.isEmpty())
+			result.add(stack.pop());
+		
+		return result;
+	}
+	
+	/**
+	 * Depth first search algorithm implementation used in the sort method
+	 * @param unsortedAlphabet - an unordered alphabet in the form of a DAG
+	 * @param marked - a map stating the state of each vertex in the DAG (BLACK, WHITE or GREY)
+	 * @param result - the list containing the ordered alphabet
+	 * @param c - the vertex currently browsed
+	 */
+	public void depthFirstSearch(Graph unsortedAlphabet, Map<Character, VertexState> marked, Stack<Character>result, char c){
+		marked.put(c, VertexState.GREY);
+		for (Character neighbour : unsortedAlphabet.getContent().get(c)) {
+			if (marked.get(neighbour) == VertexState.WHITE) {
+				this.depthFirstSearch(unsortedAlphabet, marked, result, neighbour);
+			}
+		}
+		
+		marked.put(c, VertexState.BLACK);
+		result.add(c);
+	}
+	
+
+	/**
+	 * compare 2 words and (sometimes) add their characters to an unordered alphabet (attribute unknownAlphabet)
+	 * A character is added in the unordered alphabet when 2 char are compared and:
+	 * - they differ
+	 * - they are not part of the alphabet yet
+	 * @param word1 - first word to compare
+	 * @param word2 - first word to compare
+	 */
+	public void compareWords(String word1, String word2) {
+		if (word1.isEmpty() && word2.isEmpty()) {
+			throw new IllegalArgumentException("one of the words is empty");
+		}
+		
+		int sizeWord1 = word1.length();
+		int sizeWord2 = word2.length();
+		boolean firstDifference = true;
+		int longestWord;
+		int i = 0;
+		
+		for (; i < Math.min(sizeWord1, sizeWord2); i++) {
+			char charWord1 = word1.charAt(i);
+			char charWord2 = word2.charAt(i);
+			if (!this.compareChar(charWord1, charWord2, firstDifference)) {
+				firstDifference = false;
+			}
+		}
+		
+		// the last letters of the longest word has to be part of the alphabet
+		if (sizeWord1 > sizeWord2) { longestWord = 1;}
+		else {longestWord = 2;}
+		
+		if (longestWord == 1) this.browseLastLettersOfLongest(word1, sizeWord1, i);
+		else if (longestWord == 2) this.browseLastLettersOfLongest(word2, sizeWord2, i);
+		
+		
+	
+	}
+	
+	/**
+	 * compare 2 characters. If they differ, then it is possible to order them
+	 * the first one is the one coming first in the given alphabet
+	 * This method updates the edges for the alphabet being built
+	 * @param c1 - the character coming from the first word
+	 * @param c2 - the character coming from the second word
+	 * @param firstDifference - true if it is the first time one encounters a difference in characters btwn the 2 words
+	 */
```

Comment:
You should also add @return  to make it easier to understand the return values of the method. I am a bit confused about this method's use in the if statement.